app/src/rgb_underglow.c

```diff
@@ -468,17 +468,28 @@ int zmk_rgb_underglow_change_spd(int direction) {
 
 #if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_IDLE) ||                                          \
     IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_USB)
-static int rgb_underglow_auto_state(bool *prev_state, bool new_state) {
-    if (state.on == new_state) {
+struct rgb_underglow_sleep_state {
+    bool sleeping;
+    bool pre_sleep;
+};
+
```

Comment:
The mix of the *sleep* state and the *rgb* state as bools with similar naming makes this hard to easily scan and understand.

I would  rather something like:

```suggestion
struct rgb_underglow_sleep_state {
    bool is_awake;
    bool rgb_state_before_sleeping;
};

```

See my other comment about inconsistent use of "awake" bool verses "sleep" bool to track/describe things.